web/packages/teleport/src/Account/StatePill.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Text } from 'design';
+import React from 'react';
+import styled, { css } from 'styled-components';
+
+/** State of an authentication method (password, MFA method, or passkey). */
+type State = 'active' | 'inactive';
+
+/**
+ * Renders a pill component that represents a state of an authentication method.
+ * The `state` property is both an enum value, as well as the UI text.
+ */
+export function StatePill({ state }: { state: State }) {
+  return (
+    <StatePillBody state={state}>
+      <StatePillText typography="body1">{state}</StatePillText>
+    </StatePillBody>
+  );
+}
+
+const StatePillBody = styled.span<{ backgroundColor: string; color: string }>`
+  display: inline-block;
+  vertical-align: baseline;
```

Comment:
```suggestion
  vertical-align: middle;
```

The designs actually center align instead of baseline. I confirmed with Kenny this is what we want to do.

Reply:
Oh! That's surprising, but ok.

Reply:
It actually turned out not to do anything. I assumed that the reason why you highlighted it was that the alignment was slightly off — it turned out to be a byproduct of the typography class having a custom line height that was exactly 1px too short. I decided to get rid of the Text component altogether and set the font size explicitly. This also allowed me to simplify the pill component a bit.